[Bug]: Latest vllm docker can not serve nvfp4 models. `curand_kernel.h: No such file or directory`

### Your current environment

<details>
<summary>The output of <code>python collect_env.py</code>. Just the env info of my local machine, not the docker</summary>

```text
--2025-11-27 16:36:01--  https://raw.githubusercontent.com/vllm-project/vllm/main/vllm/collect_env.py
Resolving raw.githubusercontent.com (raw.githubusercontent.com)... 2606:50c0:8000::154, 2606:50c0:8001::154, 2606:50c0:8002::154, ...
Connecting to raw.githubusercontent.com (raw.githubusercontent.com)|2606:50c0:8000::154|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 28050 (27K) [text/plain]
Saving to: ‘collect_env.py’

collect_env.py                    100%[==========================================================>]  27.39K  --.-KB/s    in 0.008s

2025-11-27 16:36:02 (3.44 MB/s) - ‘collect_env.py’ saved [28050/28050]

(vllm) luoxiao@LX-HOME:/mnt/g/download$ python collect_env.py
Collecting environment information...
==============================
        System Info
==============================
OS                           : Ubuntu 24.04.3 LTS (x86_64)
GCC version                  : (Ubuntu 13.3.0-6ubuntu2~24.04) 13.3.0
Clang version                : Could not collect
CMake version                : Could not collect
Libc version                 : glibc-2.39

==============================
       PyTorch Info
==============================
PyTorch version              : 2.9.0+cu128
Is debug build               : False
CUDA used to build PyTorch   : 12.8
ROCM used to build PyTorch   : N/A

==============================
      Python Environment
==============================
Python version               : 3.12.12 | packaged by conda-forge | (main, Oct 22 2025, 23:25:55) [GCC 14.3.0] (64-bit runtime)
Python platform              : Linux-6.6.87.2-microsoft-standard-WSL2-x86_64-with-glibc2.39

==============================
       CUDA / GPU Info
==============================
Is CUDA available            : True
CUDA runtime version         : 13.0.88
CUDA_MODULE_LOADING set to   :
GPU models and configuration : GPU 0: NVIDIA GeForce RTX 5090
Nvidia driver version        : 581.80
cuDNN version                : Could not collect
HIP runtime version          : N/A
MIOpen runtime version       : N/A
Is XNNPACK available         : True

==============================
          CPU Info
==============================
Architecture:                         x86_64
CPU op-mode(s):                       32-bit, 64-bit
Address sizes:                        48 bits physical, 48 bits virtual
Byte Order:                           Little Endian
CPU(s):                               32
On-line CPU(s) list:                  0-31
Vendor ID:                            AuthenticAMD
Model name:                           AMD Ryzen 9 9950X 16-Core Processor
CPU family:                           26
Model:                                68
Thread(s) per core:                   2
Core(s) per socket:                   16
Socket(s):                            1
Stepping:                             0
BogoMIPS:                             8599.83
Flags:                                fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush mmx fxsr sse sse2 ht syscall nx mmxext fxsr_opt pdpe1gb rdtscp lm constant_tsc rep_good nopl tsc_reliable nonstop_tsc cpuid extd_apicid tsc_known_freq pni pclmulqdq ssse3 fma cx16 sse4_1 sse4_2 movbe popcnt aes xsave avx f16c rdrand hypervisor lahf_lm cmp_legacy svm cr8_legacy abm sse4a misalignsse 3dnowprefetch osvw topoext perfctr_core ssbd ibrs ibpb stibp vmmcall fsgsbase bmi1 avx2 smep bmi2 erms invpcid avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves avx_vnni avx512_bf16 clzero xsaveerptr arat npt nrip_save tsc_scale vmcb_clean flushbyasid decodeassists pausefilter pfthreshold v_vmsave_vmload avx512vbmi umip avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg avx512_vpopcntdq rdpid fsrm avx512_vp2intersect
Virtualization:                       AMD-V
Hypervisor vendor:                    Microsoft
Virtualization type:                  full
L1d cache:                            768 KiB (16 instances)
L1i cache:                            512 KiB (16 instances)
L2 cache:                             16 MiB (16 instances)
L3 cache:                             32 MiB (1 instance)
NUMA node(s):                         1
NUMA node0 CPU(s):                    0-31
Vulnerability Gather data sampling:   Not affected
Vulnerability Itlb multihit:          Not affected
Vulnerability L1tf:                   Not affected
Vulnerability Mds:                    Not affected
Vulnerability Meltdown:               Not affected
Vulnerability Mmio stale data:        Not affected
Vulnerability Reg file data sampling: Not affected
Vulnerability Retbleed:               Not affected
Vulnerability Spec rstack overflow:   Not affected
Vulnerability Spec store bypass:      Mitigation; Speculative Store Bypass disabled via prctl
Vulnerability Spectre v1:             Mitigation; usercopy/swapgs barriers and __user pointer sanitization
Vulnerability Spectre v2:             Mitigation; Retpolines; IBPB conditional; IBRS_FW; STIBP always-on; RSB filling; PBRSB-eIBRS Not affected; BHI Not affected
Vulnerability Srbds:                  Not affected
Vulnerability Tsx async abort:        Not affected

==============================
Versions of relevant libraries
==============================
[pip3] flashinfer-python==0.5.2
[pip3] numpy==2.2.6
[pip3] nvidia-cublas-cu12==12.8.4.1
[pip3] nvidia-cuda-cupti-cu12==12.8.90
[pip3] nvidia-cuda-nvrtc-cu12==12.8.93
[pip3] nvidia-cuda-runtime-cu12==12.8.90
[pip3] nvidia-cudnn-cu12==9.10.2.21
[pip3] nvidia-cudnn-frontend==1.16.0
[pip3] nvidia-cufft-cu12==11.3.3.83
[pip3] nvidia-cufile-cu12==1.13.1.3
[pip3] nvidia-curand-cu12==10.3.9.90
[pip3] nvidia-cusolver-cu12==11.7.3.90
[pip3] nvidia-cusparse-cu12==12.5.8.93
[pip3] nvidia-cusparselt-cu12==0.7.1
[pip3] nvidia-cutlass-dsl==4.3.0
[pip3] nvidia-ml-py==13.580.82
[pip3] nvidia-nccl-cu12==2.27.5
[pip3] nvidia-nvjitlink-cu12==12.8.93
[pip3] nvidia-nvshmem-cu12==3.3.20
[pip3] nvidia-nvtx-cu12==12.8.90
[pip3] pyzmq==27.1.0
[pip3] torch==2.9.0
[pip3] torchaudio==2.9.0
[pip3] torchvision==0.24.0
[pip3] transformers==4.57.3
[pip3] triton==3.5.0
[conda] flashinfer-python                    0.5.2            pypi_0                pypi
[conda] numpy                                2.2.6            pypi_0                pypi
[conda] nvidia-cublas-cu12                   12.8.4.1         pypi_0                pypi
[conda] nvidia-cuda-cupti-cu12               12.8.90          pypi_0                pypi
[conda] nvidia-cuda-nvrtc-cu12               12.8.93          pypi_0                pypi
[conda] nvidia-cuda-runtime-cu12             12.8.90          pypi_0                pypi
[conda] nvidia-cudnn-cu12                    9.10.2.21        pypi_0                pypi
[conda] nvidia-cudnn-frontend                1.16.0           pypi_0                pypi
[conda] nvidia-cufft-cu12                    11.3.3.83        pypi_0                pypi
[conda] nvidia-cufile-cu12                   1.13.1.3         pypi_0                pypi
[conda] nvidia-curand-cu12                   10.3.9.90        pypi_0                pypi
[conda] nvidia-cusolver-cu12                 11.7.3.90        pypi_0                pypi
[conda] nvidia-cusparse-cu12                 12.5.8.93        pypi_0                pypi
[conda] nvidia-cusparselt-cu12               0.7.1            pypi_0                pypi
[conda] nvidia-cutlass-dsl                   4.3.0            pypi_0                pypi
[conda] nvidia-ml-py                         13.580.82        pypi_0                pypi
[conda] nvidia-nccl-cu12                     2.27.5           pypi_0                pypi
[conda] nvidia-nvjitlink-cu12                12.8.93          pypi_0                pypi
[conda] nvidia-nvshmem-cu12                  3.3.20           pypi_0                pypi
[conda] nvidia-nvtx-cu12                     12.8.90          pypi_0                pypi
[conda] pyzmq                                27.1.0           pypi_0                pypi
[conda] torch                                2.9.0            pypi_0                pypi
[conda] torchaudio                           2.9.0            pypi_0                pypi
[conda] torchvision                          0.24.0           pypi_0                pypi
[conda] transformers                         4.57.3           pypi_0                pypi
[conda] triton                               3.5.0            pypi_0                pypi

==============================
         vLLM Info
==============================
ROCM Version                 : Could not collect
vLLM Version                 : 0.11.2
vLLM Build Flags:
  CUDA Archs: Not Set; ROCm: Disabled
GPU Topology:
        GPU0    CPU Affinity    NUMA Affinity   GPU NUMA ID
GPU0     X                              N/A

Legend:

  X    = Self
  SYS  = Connection traversing PCIe as well as the SMP interconnect between NUMA nodes (e.g., QPI/UPI)
  NODE = Connection traversing PCIe as well as the interconnect between PCIe Host Bridges within a NUMA node
  PHB  = Connection traversing PCIe as well as a PCIe Host Bridge (typically the CPU)
  PXB  = Connection traversing multiple PCIe bridges (without traversing the PCIe Host Bridge)
  PIX  = Connection traversing at most a single PCIe bridge
  NV#  = Connection traversing a bonded set of # NVLinks

==============================
     Environment Variables
==============================
LD_LIBRARY_PATH=/usr/local/cuda/lib64:
PYTORCH_NVML_BASED_CUDA_CHECK=1
TORCHINDUCTOR_COMPILE_THREADS=1

```

</details>


### 🐛 Describe the bug

Main problem seems to be following:
```
/usr/local/lib/python3.12/dist-packages/flashinfer/data/cutlass/tools/util/include/cutlass/util/reference/device/tensor_fill.h:52:10: fatal error: curand_kernel.h: No such file or directory
```

I can run the model via my local env, where the model is quantized by following the example https://github.com/vllm-project/llm-compressor/blob/main/examples/quantization_w4a4_fp4/qwen3_vl_moe_w4a4_fp4.py

Docker image is the latest one: `vllm/vllm-openai:nightly-11ea5ec1ff7afc5ba181cba41f0cc2e4053e27f3`

<details>
<summary>Detailed log</summary>

```
vllm-nemotron9b-1  | (EngineCore_DP0 pid=143) ERROR 11-26 21:20:36 [core.py:842]   File "/usr/local/lib/python3.12/dist-packages/flashinfer/jit/core.py", line 309, in build_and_load
vllm-nemotron9b-1  | (EngineCore_DP0 pid=143) ERROR 11-26 21:20:36 [core.py:842]     self.build(verbose, need_lock=False)
vllm-nemotron9b-1  | (EngineCore_DP0 pid=143) ERROR 11-26 21:20:36 [core.py:842]   File "/usr/local/lib/python3.12/dist-packages/flashinfer/jit/core.py", line 295, in build
vllm-nemotron9b-1  | (EngineCore_DP0 pid=143) ERROR 11-26 21:20:36 [core.py:842]     run_ninja(jit_env.FLASHINFER_JIT_DIR, self.ninja_path, verbose)
vllm-nemotron9b-1  | (EngineCore_DP0 pid=143) ERROR 11-26 21:20:36 [core.py:842]   File "/usr/local/lib/python3.12/dist-packages/flashinfer/jit/cpp_ext.py", line 299, in run_ninja
vllm-nemotron9b-1  | (EngineCore_DP0 pid=143) ERROR 11-26 21:20:36 [core.py:842]     raise RuntimeError(msg) from e
vllm-nemotron9b-1  | (EngineCore_DP0 pid=143) ERROR 11-26 21:20:36 [core.py:842] RuntimeError: Ninja build failed. Ninja output:
vllm-nemotron9b-1  | (EngineCore_DP0 pid=143) ERROR 11-26 21:20:36 [core.py:842] ninja: Entering directory `/root/.cache/flashinfer/0.5.2/120a/cached_ops'
vllm-nemotron9b-1  | (EngineCore_DP0 pid=143) ERROR 11-26 21:20:36 [core.py:842] [1/4] /usr/local/cuda/bin/nvcc --generate-dependencies-with-compile --dependency-output fp4_gemm_cutlass_sm120/fp4_gemm_cutlass_sm120.cuda.o.d -DPy_LIMITED_API=0x03090000 -D_GLIBCXX_USE_CXX11_ABI=1 -isystem /usr/include/python3.12 -isystem /usr/local/cuda/include -isystem /usr/local/cuda/include/cccl -isystem /usr/local/lib/python3.12/dist-packages/tvm_ffi/include -isystem /usr/local/lib/python3.12/dist-packages/tvm_ffi/include -isystem /usr/local/lib/python3.12/dist-packages/flashinfer/data/include -isystem /usr/local/lib/python3.12/dist-packages/flashinfer/data/csrc -isystem /usr/local/lib/python3.12/dist-packages/flashinfer/data/cutlass/include -isystem /usr/local/lib/python3.12/dist-packages/flashinfer/data/cutlass/tools/util/include -isystem /usr/local/lib/python3.12/dist-packages/flashinfer/data/spdlog/include --compiler-options=-fPIC --expt-relaxed-constexpr -static-global-template-stub=false -DFLASHINFER_ENABLE_FP8_E8M0 -DFLASHINFER_ENABLE_FP4_E2M1 -std=c++17 --threads=1 -use_fast_math -DFLASHINFER_ENABLE_F16 -DFLASHINFER_ENABLE_BF16 -DFLASHINFER_ENABLE_FP8_E4M3 -DFLASHINFER_ENABLE_FP8_E5M2 -DNDEBUG -O3 -gencode=arch=compute_120a,code=sm_120a -DFLASHINFER_ENABLE_FP8_E8M0 -DFLASHINFER_ENABLE_FP4_E2M1 -DENABLE_BF16 -DENABLE_FP4 -c /usr/local/lib/python3.12/dist-packages/flashinfer/data/csrc/fp4_gemm_cutlass_sm120.cu -o fp4_gemm_cutlass_sm120/fp4_gemm_cutlass_sm120.cuda.o
vllm-nemotron9b-1  | (EngineCore_DP0 pid=143) ERROR 11-26 21:20:36 [core.py:842] FAILED: [code=1] fp4_gemm_cutlass_sm120/fp4_gemm_cutlass_sm120.cuda.o
vllm-nemotron9b-1  | (EngineCore_DP0 pid=143) ERROR 11-26 21:20:36 [core.py:842] /usr/local/cuda/bin/nvcc --generate-dependencies-with-compile --dependency-output fp4_gemm_cutlass_sm120/fp4_gemm_cutlass_sm120.cuda.o.d -DPy_LIMITED_API=0x03090000 -D_GLIBCXX_USE_CXX11_ABI=1 -isystem /usr/include/python3.12 -isystem /usr/local/cuda/include -isystem /usr/local/cuda/include/cccl -isystem /usr/local/lib/python3.12/dist-packages/tvm_ffi/include -isystem /usr/local/lib/python3.12/dist-packages/tvm_ffi/include -isystem /usr/local/lib/python3.12/dist-packages/flashinfer/data/include -isystem /usr/local/lib/python3.12/dist-packages/flashinfer/data/csrc -isystem /usr/local/lib/python3.12/dist-packages/flashinfer/data/cutlass/include -isystem /usr/local/lib/python3.12/dist-packages/flashinfer/data/cutlass/tools/util/include -isystem /usr/local/lib/python3.12/dist-packages/flashinfer/data/spdlog/include --compiler-options=-fPIC --expt-relaxed-constexpr -static-global-template-stub=false -DFLASHINFER_ENABLE_FP8_E8M0 -DFLASHINFER_ENABLE_FP4_E2M1 -std=c++17 --threads=1 -use_fast_math -DFLASHINFER_ENABLE_F16 -DFLASHINFER_ENABLE_BF16 -DFLASHINFER_ENABLE_FP8_E4M3 -DFLASHINFER_ENABLE_FP8_E5M2 -DNDEBUG -O3 -gencode=arch=compute_120a,code=sm_120a -DFLASHINFER_ENABLE_FP8_E8M0 -DFLASHINFER_ENABLE_FP4_E2M1 -DENABLE_BF16 -DENABLE_FP4 -c /usr/local/lib/python3.12/dist-packages/flashinfer/data/csrc/fp4_gemm_cutlass_sm120.cu -o fp4_gemm_cutlass_sm120/fp4_gemm_cutlass_sm120.cuda.o
vllm-nemotron9b-1  | (EngineCore_DP0 pid=143) ERROR 11-26 21:20:36 [core.py:842] In file included from /usr/local/lib/python3.12/dist-packages/flashinfer/data/include/flashinfer/cutlass_utils.cuh:42,
vllm-nemotron9b-1  | (EngineCore_DP0 pid=143) ERROR 11-26 21:20:36 [core.py:842]                  from /usr/local/lib/python3.12/dist-packages/flashinfer/data/include/flashinfer/gemm/fp4_gemm_template_sm120.h:35,
vllm-nemotron9b-1  | (EngineCore_DP0 pid=143) ERROR 11-26 21:20:36 [core.py:842]                  from /usr/local/lib/python3.12/dist-packages/flashinfer/data/include/flashinfer/gemm/fp4_gemm_cutlass_template_sm120.h:41,
vllm-nemotron9b-1  | (EngineCore_DP0 pid=143) ERROR 11-26 21:20:36 [core.py:842]                  from /usr/local/lib/python3.12/dist-packages/flashinfer/data/csrc/fp4_gemm_cutlass_sm120.cu:26:
vllm-nemotron9b-1  | (EngineCore_DP0 pid=143) ERROR 11-26 21:20:36 [core.py:842] /usr/local/lib/python3.12/dist-packages/flashinfer/data/cutlass/tools/util/include/cutlass/util/reference/device/tensor_fill.h:52:10: fatal error: curand_kernel.h: No such file or directory
vllm-nemotron9b-1  | (EngineCore_DP0 pid=143) ERROR 11-26 21:20:36 [core.py:842]    52 | #include <curand_kernel.h>
vllm-nemotron9b-1  | (EngineCore_DP0 pid=143) ERROR 11-26 21:20:36 [core.py:842]       |          ^~~~~~~~~~~~~~~~~
vllm-nemotron9b-1  | (EngineCore_DP0 pid=143) ERROR 11-26 21:20:36 [core.py:842] compilation terminated.
vllm-nemotron9b-1  | (EngineCore_DP0 pid=143) ERROR 11-26 21:20:36 [core.py:842] [2/4] /usr/local/cuda/bin/nvcc --generate-dependencies-with-compile --dependency-output fp4_gemm_cutlass_sm120/fp4_gemm_cutlass___nv_bfloat16_128_128_128.cuda.o.d -DPy_LIMITED_API=0x03090000 -D_GLIBCXX_USE_CXX11_ABI=1 -isystem /usr/include/python3.12 -isystem /usr/local/cuda/include -isystem /usr/local/cuda/include/cccl -isystem /usr/local/lib/python3.12/dist-packages/tvm_ffi/include -isystem /usr/local/lib/python3.12/dist-packages/tvm_ffi/include -isystem /usr/local/lib/python3.12/dist-packages/flashinfer/data/include -isystem /usr/local/lib/python3.12/dist-packages/flashinfer/data/csrc -isystem /usr/local/lib/python3.12/dist-packages/flashinfer/data/cutlass/include -isystem /usr/local/lib/python3.12/dist-packages/flashinfer/data/cutlass/tools/util/include -isystem /usr/local/lib/python3.12/dist-packages/flashinfer/data/spdlog/include --compiler-options=-fPIC --expt-relaxed-constexpr -static-global-template-stub=false -DFLASHINFER_ENABLE_FP8_E8M0 -DFLASHINFER_ENABLE_FP4_E2M1 -std=c++17 --threads=1 -use_fast_math -DFLASHINFER_ENABLE_F16 -DFLASHINFER_ENABLE_BF16 -DFLASHINFER_ENABLE_FP8_E4M3 -DFLASHINFER_ENABLE_FP8_E5M2 -DNDEBUG -O3 -gencode=arch=compute_120a,code=sm_120a -DFLASHINFER_ENABLE_FP8_E8M0 -DFLASHINFER_ENABLE_FP4_E2M1 -DENABLE_BF16 -DENABLE_FP4 -c /root/.cache/flashinfer/0.5.2/120a/generated/gen_gemm_sm120_cutlass_fp4/fp4_gemm_cutlass___nv_bfloat16_128_128_128.cu -o fp4_gemm_cutlass_sm120/fp4_gemm_cutlass___nv_bfloat16_128_128_128.cuda.o
vllm-nemotron9b-1  | (EngineCore_DP0 pid=143) ERROR 11-26 21:20:36 [core.py:842] FAILED: [code=1] fp4_gemm_cutlass_sm120/fp4_gemm_cutlass___nv_bfloat16_128_128_128.cuda.o
vllm-nemotron9b-1  | (EngineCore_DP0 pid=143) ERROR 11-26 21:20:36 [core.py:842] /usr/local/cuda/bin/nvcc --generate-dependencies-with-compile --dependency-output fp4_gemm_cutlass_sm120/fp4_gemm_cutlass___nv_bfloat16_128_128_128.cuda.o.d -DPy_LIMITED_API=0x03090000 -D_GLIBCXX_USE_CXX11_ABI=1 -isystem /usr/include/python3.12 -isystem /usr/local/cuda/include -isystem /usr/local/cuda/include/cccl -isystem /usr/local/lib/python3.12/dist-packages/tvm_ffi/include -isystem /usr/local/lib/python3.12/dist-packages/tvm_ffi/include -isystem /usr/local/lib/python3.12/dist-packages/flashinfer/data/include -isystem /usr/local/lib/python3.12/dist-packages/flashinfer/data/csrc -isystem /usr/local/lib/python3.12/dist-packages/flashinfer/data/cutlass/include -isystem /usr/local/lib/python3.12/dist-packages/flashinfer/data/cutlass/tools/util/include -isystem /usr/local/lib/python3.12/dist-packages/flashinfer/data/spdlog/include --compiler-options=-fPIC --expt-relaxed-constexpr -static-global-template-stub=false -DFLASHINFER_ENABLE_FP8_E8M0 -DFLASHINFER_ENABLE_FP4_E2M1 -std=c++17 --threads=1 -use_fast_math -DFLASHINFER_ENABLE_F16 -DFLASHINFER_ENABLE_BF16 -DFLASHINFER_ENABLE_FP8_E4M3 -DFLASHINFER_ENABLE_FP8_E5M2 -DNDEBUG -O3 -gencode=arch=compute_120a,code=sm_120a -DFLASHINFER_ENABLE_FP8_E8M0 -DFLASHINFER_ENABLE_FP4_E2M1 -DENABLE_BF16 -DENABLE_FP4 -c /root/.cache/flashinfer/0.5.2/120a/generated/gen_gemm_sm120_cutlass_fp4/fp4_gemm_cutlass___nv_bfloat16_128_128_128.cu -o fp4_gemm_cutlass_sm120/fp4_gemm_cutlass___nv_bfloat16_128_128_128.cuda.o
vllm-nemotron9b-1  | (EngineCore_DP0 pid=143) ERROR 11-26 21:20:36 [core.py:842] In file included from /usr/local/lib/python3.12/dist-packages/flashinfer/data/include/flashinfer/cutlass_utils.cuh:42,
vllm-nemotron9b-1  | (EngineCore_DP0 pid=143) ERROR 11-26 21:20:36 [core.py:842]                  from /usr/local/lib/python3.12/dist-packages/flashinfer/data/include/flashinfer/gemm/fp4_gemm_template_sm120.h:35,
vllm-nemotron9b-1  | (EngineCore_DP0 pid=143) ERROR 11-26 21:20:36 [core.py:842]                  from /usr/local/lib/python3.12/dist-packages/flashinfer/data/include/flashinfer/gemm/fp4_gemm_cutlass_template_sm120.h:41,
vllm-nemotron9b-1  | (EngineCore_DP0 pid=143) ERROR 11-26 21:20:36 [core.py:842]                  from /root/.cache/flashinfer/0.5.2/120a/generated/gen_gemm_sm120_cutlass_fp4/fp4_gemm_cutlass___nv_bfloat16_128_128_128.cu:18:
vllm-nemotron9b-1  | (EngineCore_DP0 pid=143) ERROR 11-26 21:20:36 [core.py:842] /usr/local/lib/python3.12/dist-packages/flashinfer/data/cutlass/tools/util/include/cutlass/util/reference/device/tensor_fill.h:52:10: fatal error: curand_kernel.h: No such file or directory
vllm-nemotron9b-1  | (EngineCore_DP0 pid=143) ERROR 11-26 21:20:36 [core.py:842]    52 | #include <curand_kernel.h>
vllm-nemotron9b-1  | (EngineCore_DP0 pid=143) ERROR 11-26 21:20:36 [core.py:842]       |          ^~~~~~~~~~~~~~~~~
vllm-nemotron9b-1  | (EngineCore_DP0 pid=143) ERROR 11-26 21:20:36 [core.py:842] compilation terminated.
vllm-nemotron9b-1  | (EngineCore_DP0 pid=143) ERROR 11-26 21:20:36 [core.py:842] [3/4] /usr/local/cuda/bin/nvcc --generate-dependencies-with-compile --dependency-output fp4_gemm_cutlass_sm120/fp4_gemm_cutlass_half_128_128_128.cuda.o.d -DPy_LIMITED_API=0x03090000 -D_GLIBCXX_USE_CXX11_ABI=1 -isystem /usr/include/python3.12 -isystem /usr/local/cuda/include -isystem /usr/local/cuda/include/cccl -isystem /usr/local/lib/python3.12/dist-packages/tvm_ffi/include -isystem /usr/local/lib/python3.12/dist-packages/tvm_ffi/include -isystem /usr/local/lib/python3.12/dist-packages/flashinfer/data/include -isystem /usr/local/lib/python3.12/dist-packages/flashinfer/data/csrc -isystem /usr/local/lib/python3.12/dist-packages/flashinfer/data/cutlass/include -isystem /usr/local/lib/python3.12/dist-packages/flashinfer/data/cutlass/tools/util/include -isystem /usr/local/lib/python3.12/dist-packages/flashinfer/data/spdlog/include --compiler-options=-fPIC --expt-relaxed-constexpr -static-global-template-stub=false -DFLASHINFER_ENABLE_FP8_E8M0 -DFLASHINFER_ENABLE_FP4_E2M1 -std=c++17 --threads=1 -use_fast_math -DFLASHINFER_ENABLE_F16 -DFLASHINFER_ENABLE_BF16 -DFLASHINFER_ENABLE_FP8_E4M3 -DFLASHINFER_ENABLE_FP8_E5M2 -DNDEBUG -O3 -gencode=arch=compute_120a,code=sm_120a -DFLASHINFER_ENABLE_FP8_E8M0 -DFLASHINFER_ENABLE_FP4_E2M1 -DENABLE_BF16 -DENABLE_FP4 -c /root/.cache/flashinfer/0.5.2/120a/generated/gen_gemm_sm120_cutlass_fp4/fp4_gemm_cutlass_half_128_128_128.cu -o fp4_gemm_cutlass_sm120/fp4_gemm_cutlass_half_128_128_128.cuda.o
vllm-nemotron9b-1  | (EngineCore_DP0 pid=143) ERROR 11-26 21:20:36 [core.py:842] FAILED: [code=1] fp4_gemm_cutlass_sm120/fp4_gemm_cutlass_half_128_128_128.cuda.o
vllm-nemotron9b-1  | (EngineCore_DP0 pid=143) ERROR 11-26 21:20:36 [core.py:842] /usr/local/cuda/bin/nvcc --generate-dependencies-with-compile --dependency-output fp4_gemm_cutlass_sm120/fp4_gemm_cutlass_half_128_128_128.cuda.o.d -DPy_LIMITED_API=0x03090000 -D_GLIBCXX_USE_CXX11_ABI=1 -isystem /usr/include/python3.12 -isystem /usr/local/cuda/include -isystem /usr/local/cuda/include/cccl -isystem /usr/local/lib/python3.12/dist-packages/tvm_ffi/include -isystem /usr/local/lib/python3.12/dist-packages/tvm_ffi/include -isystem /usr/local/lib/python3.12/dist-packages/flashinfer/data/include -isystem /usr/local/lib/python3.12/dist-packages/flashinfer/data/csrc -isystem /usr/local/lib/python3.12/dist-packages/flashinfer/data/cutlass/include -isystem /usr/local/lib/python3.12/dist-packages/flashinfer/data/cutlass/tools/util/include -isystem /usr/local/lib/python3.12/dist-packages/flashinfer/data/spdlog/include --compiler-options=-fPIC --expt-relaxed-constexpr -static-global-template-stub=false -DFLASHINFER_ENABLE_FP8_E8M0 -DFLASHINFER_ENABLE_FP4_E2M1 -std=c++17 --threads=1 -use_fast_math -DFLASHINFER_ENABLE_F16 -DFLASHINFER_ENABLE_BF16 -DFLASHINFER_ENABLE_FP8_E4M3 -DFLASHINFER_ENABLE_FP8_E5M2 -DNDEBUG -O3 -gencode=arch=compute_120a,code=sm_120a -DFLASHINFER_ENABLE_FP8_E8M0 -DFLASHINFER_ENABLE_FP4_E2M1 -DENABLE_BF16 -DENABLE_FP4 -c /root/.cache/flashinfer/0.5.2/120a/generated/gen_gemm_sm120_cutlass_fp4/fp4_gemm_cutlass_half_128_128_128.cu -o fp4_gemm_cutlass_sm120/fp4_gemm_cutlass_half_128_128_128.cuda.o
vllm-nemotron9b-1  | (EngineCore_DP0 pid=143) ERROR 11-26 21:20:36 [core.py:842] In file included from /usr/local/lib/python3.12/dist-packages/flashinfer/data/include/flashinfer/cutlass_utils.cuh:42,
vllm-nemotron9b-1  | (EngineCore_DP0 pid=143) ERROR 11-26 21:20:36 [core.py:842]                  from /usr/local/lib/python3.12/dist-packages/flashinfer/data/include/flashinfer/gemm/fp4_gemm_template_sm120.h:35,
vllm-nemotron9b-1  | (EngineCore_DP0 pid=143) ERROR 11-26 21:20:36 [core.py:842]                  from /usr/local/lib/python3.12/dist-packages/flashinfer/data/include/flashinfer/gemm/fp4_gemm_cutlass_template_sm120.h:41,
vllm-nemotron9b-1  | (EngineCore_DP0 pid=143) ERROR 11-26 21:20:36 [core.py:842]                  from /root/.cache/flashinfer/0.5.2/120a/generated/gen_gemm_sm120_cutlass_fp4/fp4_gemm_cutlass_half_128_128_128.cu:18:
vllm-nemotron9b-1  | (EngineCore_DP0 pid=143) ERROR 11-26 21:20:36 [core.py:842] /usr/local/lib/python3.12/dist-packages/flashinfer/data/cutlass/tools/util/include/cutlass/util/reference/device/tensor_fill.h:52:10: fatal error: curand_kernel.h: No such file or directory
vllm-nemotron9b-1  | (EngineCore_DP0 pid=143) ERROR 11-26 21:20:36 [core.py:842]    52 | #include <curand_kernel.h>
vllm-nemotron9b-1  | (EngineCore_DP0 pid=143) ERROR 11-26 21:20:36 [core.py:842]       |          ^~~~~~~~~~~~~~~~~
vllm-nemotron9b-1  | (EngineCore_DP0 pid=143) ERROR 11-26 21:20:36 [core.py:842] compilation terminated.
vllm-nemotron9b-1  | (EngineCore_DP0 pid=143) ERROR 11-26 21:20:36 [core.py:842] ninja: build stopped: subcommand failed.
vllm-nemotron9b-1  | (EngineCore_DP0 pid=143) ERROR 11-26 21:20:36 [core.py:842]
vllm-nemotron9b-1  | (EngineCore_DP0 pid=143) Traceback (most recent call last):
vllm-nemotron9b-1  | (EngineCore_DP0 pid=143)   File "/usr/local/lib/python3.12/dist-packages/flashinfer/jit/cpp_ext.py", line 287, in run_ninja
```

</details>

### Before submitting a new issue...

- [x] Make sure you already searched for relevant issues, and asked the chatbot living at the bottom right corner of the [documentation page](https://docs.vllm.ai/en/latest/), which can answer lots of frequently asked questions.

--- a/docker/Dockerfile
+++ b/docker/Dockerfile
@@ -366,7 +366,7 @@ RUN CUDA_VERSION_DASH=$(echo $CUDA_VERSION | cut -d. -f1,2 | tr '.' '-') && \
     cuda-cuobjdump-${CUDA_VERSION_DASH} \
     # https://github.com/vllm-project/vllm/issues/29590
     libcurand-dev-${CUDA_VERSION_DASH} \
-    libcublas-${CUDA_VERSION_DASH}
+    libcublas-${CUDA_VERSION_DASH} \
     # Fixes nccl_allocator requiring nccl.h at runtime
     # https://github.com/vllm-project/vllm/blob/1336a1ea244fa8bfd7e72751cabbdb5b68a0c11a/vllm/distributed/device_communicators/pynccl_allocator.py#L22
     libnccl-dev && \

--- a/docker/Dockerfile
+++ b/docker/Dockerfile
@@ -364,7 +364,12 @@ RUN CUDA_VERSION_DASH=$(echo $CUDA_VERSION | cut -d. -f1,2 | tr '.' '-') && \
     cuda-cudart-${CUDA_VERSION_DASH} \
     cuda-nvrtc-${CUDA_VERSION_DASH} \
     cuda-cuobjdump-${CUDA_VERSION_DASH} \
-    libcublas-${CUDA_VERSION_DASH} && \
+    # https://github.com/vllm-project/vllm/issues/29590
+    libcurand-dev-${CUDA_VERSION_DASH} \
+    libcublas-${CUDA_VERSION_DASH}
+    # Fixes nccl_allocator requiring nccl.h at runtime
+    # https://github.com/vllm-project/vllm/blob/1336a1ea244fa8bfd7e72751cabbdb5b68a0c11a/vllm/distributed/device_communicators/pynccl_allocator.py#L22
+    libnccl-dev && \
     rm -rf /var/lib/apt/lists/*
 
 ARG PIP_INDEX_URL UV_INDEX_URL

